Add blurb about CodeNarc tool autodetection to release notes (#22515)
gradle/gradle: subprojects/docs/src/docs/release/notes.md

vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv -->

#### PMD and CodeNarc tasks execute in parallel by default
The [PMD](userguide/pmd_plugin.html) and [CodeNarc](userguide/codenarc_plugin.html) plugins now use the Gradle worker API and JVM toolchains. These tools now perform analysis via an external worker process and therefore their tasks may now run in parallel within one project.

In Java projects, these tools will use the same version of Java required by the project. In other types of projects, they will use the same version of Java that is used by the Gradle daemon.

#### CodeNarc Plugin automatically detects appropriate version for current Groovy runtime

> The [CodeNarc](https://codenarc.org/) project now publishes separate versions for use with Groovy 4.  Although Gradle still currently ships with Groovy 3, to ensure future compatibility the [CodeNarcPlugin](userguide/codenarc_plugin.html) will now by default automatically detect the appropriate version of CodeNarc to use for the current Groovy runtime.  The `toolVersion` property on the [CodeNarcExtension](current/dsl/org.gradle.api.plugins.quality.CodeNarcExtension.html#org.gradle.api.plugins.quality.CodeNarcExtension) can still be used to explicitly specify the version of CodeNarc to use.

Reviewer: ```suggestion
The [CodeNarc](https://codenarc.org/) project now publishes separate versions for use with Groovy 4.
Gradle still currently ships with Groovy 3.
To ensure future compatibility, the [CodeNarcPlugin](userguide/codenarc_plugin.html) now automatically detects the appropriate version of CodeNarc for the current Groovy runtime.
You can still explicitly specify a CodeNarc version with the `toolVersion` property on the [CodeNarcExtension](current/dsl/org.gradle.api.plugins.quality.CodeNarcExtension.html#org.gradle.api.plugins.quality.CodeNarcExtension).
```
Suggestion: split up some long sentences, trim a few words.

> The [CodeNarc](https://codenarc.org/) project now publishes separate versions for use with Groovy 4.  Although Gradle still currently ships with Groovy 3, to ensure future compatibility the [CodeNarcPlugin](userguide/codenarc_plugin.html) will now by default automatically detect the appropriate version of CodeNarc to use for the current Groovy runtime.  The `toolVersion` property on the [CodeNarcExtension](current/dsl/org.gradle.api.plugins.quality.CodeNarcExtension.html#org.gradle.api.plugins.quality.CodeNarcExtension) can still be used to explicitly specify the version of CodeNarc to use.

Reviewer 2: This shouldn't be linking to /current/
